spec/abci++/abci++_basic_concepts_002_draft.md

```diff
@@ -1,51 +1,266 @@
 
-### **Consensus** connection
+ABCI++ overcomes these limitations by allowing the application to interfere at three key places of the block execution. The new interface allows block proposers to perform application-dependent work in a block through the `PrepareProposal` method; validators to perform application-dependent work in a proposed block through the `ProcessProposal` method; and applications to require their validators do more than just validate blocks, e.i., validator guaranteed IBC connection attempts, through the `ExtendVote` and `VerifyVoteExtension` methods. Furthermore, ABCI++ renames {`BeginBlock`, [`DeliverTx`], `EndBlock`} to `FinalizeBlock`, as the new methods makes them no longer useful.
 
-* Driven by a consensus protocol and is responsible for block execution.
-* Handles the `InitChain`, `PrepareProposal`, `ProcessProposal`, `ExtendVote`,
-  `VerifyVoteExtension`, and `FinalizeBlock` method calls.
+## Methods overview
+[&uparrow; Back to Outline](#outline)
 
-### **Mempool** connection
+Methods can be classified into four categories: consensus, mempool, info, and state-sync. When Tendermint and the Application run in separate processes, Tendermint opens a different connection for each category to enhance parallelism.
 
-* For validating new transactions, before they're shared or included in a block.
-* Handles the `CheckTx` calls.
+### Consensus/block execution methods
 
-### **Info** connection
+The first time a new blockchain is started, Tendermint calls
+`InitChain`. From then on, method `FinalizeBlock` is executed at the end of each
+block, resulting in an updated Application state.
+During consensus execution of a block height, before method `FinalizeBlock` is
+called, methods `PrepareProposal`, `ProcessProposal`, `ExtendVote`, and
+`VerifyVoteExtension` may be called several times.
+See [Tendermint's expected behavior](abci++_tmint_expected_behavior_002_draft.md)
+for details on the possible call sequences of these methods.
+
+* [**InitChain:**](./abci++_methods_002_draft.md#initchain) This method initializes the blockchain. Tendermint calls it once upon genesis.
+
+* [**PrepareProposal:**](./abci++_methods_002_draft.md#prepareproposal) It allows the block proposer to perform application-dependent work in a block before sending its proposal. This enables, for instance, batch optimizations to a block, which has been empirically demonstrated to be a key component for scaling. Method `PrepareProposal` is called every time Tendermint is about to send
+a proposal message, but no previous proposal has been locked at Tendermint level.
+Tendermint gathers outstanding transactions from the mempool, generates a block header, and uses
+them to create a block to propose. Then, it calls `RequestPrepareProposal`
+with the newly created proposal, called _raw block_. The Application can
```

Comment:
raw proposal?
I think we always call it raw proposal elsewhere in the text

Reply:
I see both used. Maybe I introduced the discrepancy. I'll do a pass and make sure we always use raw proposal.